### PR TITLE
chore: update @actions/artifact

### LINKED ISSUE #58: Upgrade Octokit to support fetching artifacts as soon as a single job is complete
I have several jobs that run simultaneously. Unfortunately, artifacts don't become accessible until all jobs are complete. Since the artifacts aren't available yet, a "null" commit comes back as the compare commit when I push new commits before all the tests are complete. This is because the previous artifact has been deleted and not all the jobs have been completed yet.

See the github issue here:
https://github.com/actions/upload-artifact/issues/181

This has been fixed in actions/upload-artifact so I wonder if we can do the same.

Here is the code they use to upload the file (for reference):
https://github.com/actions/upload-artifact/blob/main/src/shared/upload-artifact.ts

### PR DESCRIPTION
**Pull request by cwbusacker**
This resolves #58

### CODE REVIEW

**Comment by saurav0705** [comment]
you can skip generating files in dist i have already added a action which updates main when a commit is pushed

File: package-lock.json
Comment (saurav0705) [inline package-lock.json:1]: remove this file as we use `yarn` as package manager

File: dist/index.js
Comment (saurav0705) [inline dist/index.js:1]: remove this file

File: dist/index.js.map
Comment (saurav0705) [inline dist/index.js.map:1]: remove this file

File: dist/licenses.txt
Comment (saurav0705) [inline dist/licenses.txt:1]: remove this file

**Review by saurav0705 — CHANGES_REQUESTED** [review]

**Comment by cwbusacker** [comment]
> you can skip generating files in dist i have already added a action which updates main when a commit is pushed

I update dist so that I can test it. From now on, I'll test on my own local branch and then make the PR without the dist (that's why I left it as draft).

**Review by saurav0705 — APPROVED** [review]

**Pull request merged**